presto-parser/src/main/java/com/facebook/presto/sql/util/DoubleLiteralFormatter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.util;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+@ThreadSafe
+public final class DoubleLiteralFormatter
+{
+    private static final ThreadLocal<DecimalFormat> format = ThreadLocal.withInitial(() -> {
+        DecimalFormat formatter = ((DecimalFormat) NumberFormat.getNumberInstance(Locale.US));
```

Comment:
Doing this seems more robust, seems it avoids the downcast and prevents breakage if NumberFormat ever changes to return a different class.

```java
return new DecimalFormat("0.###################E0###", new DecimalFormatSymbols(Locale.US));
```